bigquery/client.py

```diff
@@ -327,13 +327,13 @@ def get_query_rows(self, job_id, offset=None, limit=None, timeout=0):
         records = [self._transform_row(row, schema) for row in rows]
 
         # Append to records if there are multiple pages for query results
-        while page_token:
+        while limit is None and page_token or len(records) < limit:
```

Comment:
I believe this can be `while page_token and (not limit or len(records) < limit)`. I think this would fix the failing unit test.